Fix off-by-one error in check for future delta layers.

hlinnaka opened the pull request:
This doesnt show up at the moment, because we never create a delta
layer with end-LSN equal to the last LSN. We always create an image
layer at that LSN instead. For example, if the latest processed LSN is
100, we would create a delta layer with end LSN 100 (exclusive), and
an image layer at 100. But that's just how InMemoryLayer::write_to_disk
happens to work at the moment, there's no fundamental reason it needs
to always create that image layer. I noticed this bug when I tried to
change the logic in InMemoryLayer::write_to_disk to only create an
image layer after a few delta layers.

LizardWizzard commented:
I see a bunch of postgres changes, is it intentional? 

Will it be hard to automatically test this too?

hlinnaka commented:
> I see a bunch of postgres changes, is it intentional?

Nope, reverted.

> Will it be hard to automatically test this too?

Added a unit test.

Thanks!

LizardWizzard reviewed (approved):
Great! Thanks!
